app/controllers/users_controller.rb

```diff
@@ -15,21 +15,35 @@ def edit
 
   def update
     @user = User.find params[:id]
-    if can?(:manage, User)
+    if @user.update user_params
```

Comment:
Remove this role updating part with the `role_adder` because #update is now only responsible for updating password